src/main/java/org/apache/maven/plugins/enforcer/EnforceBytecodeVersion.java

```diff
@@ -138,6 +140,12 @@ static String renderVersion( int major, int minor )
      */
     private String[] ignoredScopes;
 
+    /**
+     * Ignore all dependencies which have {@code &lt;optional&gt;true&lt;/optional&gt;}.
+     * @since 1.0.0
+     */
+    private boolean ignoreOptionals = false;
+

```

Comment:
IIUC, this addition and the three lines way below with the `if` are the actual intended change here.

Also, it would be warmly welcome to write an additional dedicated IT (see src/it directory).